File: app/src/test/java/com/kelsos/mbrc/rules/DBFlowTestRule.java
package com.kelsos.mbrc.rules;

import android.support.annotation.NonNull;

import com.raizlabs.android.dbflow.config.FlowConfig;
import com.raizlabs.android.dbflow.config.FlowManager;

import org.junit.rules.TestRule;
import org.junit.runner.Description;
import org.junit.runners.model.Statement;
import org.robolectric.RuntimeEnvironment;

public class DBFlowTestRule implements TestRule {
  @Override
  public Statement apply(Statement base, Description description) {
    return new Statement() {
      @Override
      public void evaluate() throws Throwable {
        FlowManager.init(new FlowConfig.Builder(RuntimeEnvironment.application).build());
        try {
          base.evaluate();
        } finally {
          FlowManager.reset();
          FlowManager.destroy();
        }
      }
    };
  }

  @NonNull public static DBFlowTestRule create() {
    return new DBFlowTestRule();
  }
}

File: app/src/test/java/com/kelsos/mbrc/rules/DBFlowTestRule.kt
package com.kelsos.mbrc.rules

import com.raizlabs.android.dbflow.config.FlowConfig
import com.raizlabs.android.dbflow.config.FlowManager
import org.junit.rules.TestRule
import org.junit.runner.Description
import org.junit.runners.model.Statement
import org.robolectric.RuntimeEnvironment

class DBFlowTestRule : TestRule {
  override fun apply(
    base: Statement,
    description: Description,
  ): Statement =
    object : Statement() {
      @Throws(Throwable::class)
      override fun evaluate() {
        FlowManager.init(FlowConfig.Builder(RuntimeEnvironment.application).build())
        try {
          base.evaluate()
        } finally {
          FlowManager.reset()
          FlowManager.destroy()
        }
      }
    }

  companion object {
    fun create(): DBFlowTestRule = DBFlowTestRule()
  }
}
